### ISSUE TITLE
RCS1163 shouldn't trigger for delegates

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#342]
```csharp
class Demo
{
    public void Dee(Action<string, int> eee) {  }
}

class Program
{
    static void Main(string[] args)
    {
        var z = new Demo();
        z.Dee((str, integer) => Console.WriteLine(integer));
    }
}
```

RCS1163 (or at least, the analyzer part) triggers on the `str` variable, even though that variable can't be removed because doing so would result in uncompilable code.

Additionally, when RCS1163 triggers on such a variable, the codefix cannot be selected - the only option available is to suppress this analyzer.

**Comment by JosefPihrt** [comment]
Analyzer is right that the parameter is not used.

Recommended approach in this case is to rename unused analyzer to "_". Then it will not be reported.

This analyzer does not provide a code fix and rather relies on built-in "Change signature..." dialog.
